File: JavaBlock/src/config/scriptInit.py
#!/usr/bin/python

from math import *
import random
def Write(a):
    InputReader.Write(str(a))
def Writeln(a):
    InputReader.Write(a+'\n')

def Print(a):
    InputReader.Write(str(a))
def Println(a):
    InputReader.Write(a+'\n')
def Read(str="Input string:"):
    if len(str)==0: str="Input string:"
    return InputReader.readString(str)
def ReadM(str="Input string"):
    if len(str)==0: str="Input string:"
    return InputReader.readString(str)
def ReadNumber(str="Input number"):
    if len(str)==0: str="Input number"
    return float(InputReader.readNumber(str))
def ReadInteger(str="Input integer:"):
    if len(str)==0: str="Input integer:"
    return int(InputReader.readNumber(str))
def ReadLogic(str="Input Logic:"):
    if len(str)==0: str="Input Logic:"
    return bool(InputReader.readLogic(str))
def ReadCharArray(str="Input char array:"):
    if len(str)==0: str="Input char array:"
    s=Read(str)
    t=[]
    for C in s:
        t.append(ord(C))
    return t

def rand(f, t):
    return random.randint(f,t)

def round(a):
    if a-floor(a)>=0.5:
        return ceil(a)
    return floor(a)

def round(a, p=1):
    a/=p;
    if abs(a-floor(a))>=0.5:
      a=ceil(a)
    else:
      a=floor(a)
    return a*p

PI=pi
def toRadians(a):
    return (a/180)*PI
def toDegrees(a):
    return (a/PI)*180

def getValueOf(a):
    if a.__class__=="".__class__:
        return a
    if a.__class__==True.__class__:
        if(a==True): return str('T')
        else: return str('F')
    if a.__class__==[].__class__:
        Sret123="[";
        for i in range(len(a)):
            Sret123+=getValueOf(a[i])
            if i<len(a)-1: Sret123+=","
        Sret123+="]"
        return Sret123
    return str(a)

def Array(n):
    TAB=[]
    while n>0:
        TAB.append(0);
        n-=1
    return TAB
def stringToArray(s):
    T=[]
    for i in range(len(s)):
        T.append=ord(s[i])
    return T
def arrayToString(s):
    T=""
    for i in s:
        T+=chr(i)
    return T
def toChar(x):
    return char(x)
def newArray(n):
    TAB=[]
    while n>0:
        TAB.append(0);
        n-=1
    return TAB


def toInt(a):
    return int(a)
def toFloat(a):
    return float(a)
def toLogic(a):
    if a==True: return True
    if a=="true": return True
    if a!=0: return True
    return False;

true=True
false=False

File: JavaBlock/src/config/scriptInit.js
function Write(a){
    //outStream.append(a)
    InputReader.Write(a)
}
function Writeln(a){
    //outStream.append(a+'\n')
    InputReader.Write(a+'\n')
}
function Print(a){
    //outStream.append(a)
    InputReader.Write(a)
}
function Println(a){
    //outStream.append(a+'\n')
    InputReader.Write(a+'\n')
}
function Read(){
    return InputReader.readString("Input")
}
function Read(m){
    return ReadM(m)
}
function ReadM(message){
    if(typeof(message)!="string") message="Input"
    var jstr1=InputReader.readString(message)
    return ""+jstr1
}

function ReadNumber(m){
    if(typeof(m)!="string") m="Input number"
    return InputReader.readNumber(m)
}
function ReadInteger(m){
    if(typeof(m)!="string") m="Input integer"
    return InputReader.readInteger(m)
}
function ReadLogic(m){
    if(typeof(m)!="string") m="Input true/false"
    return InputReader.readLogic(m)
}
function ReadCharArray(m){
    if(typeof(m)!="string") m="Input char array"
    return InputReader.readCharArray(m)
}

function stringToArray(str){
    var len=str.length
    var tmp=new Array(len)
    if(!(len>=0))
        len=str.length
    for(i=0; i<len; i++){
        tmp[i]=str.charCodeAt(i)
    }
    return tmp
}
function arrayToString(arr){
    var str=""
    for(i=0; i<arr.length; i++){
        str+=String.fromCharCode(arr[i])
    }
    return str
}

function ReadCharArray(){
    tmp=InputReader.readString("input string")
    return stringToArray(tmp)
}
function fromCharCode(c){
    String.fromCharCode(c)
}
function charCodeAt(c){
    String.charCodeAt(c)
}
PI=Math.PI
pi=PI
function toRadians(a){return (a/180)*PI}
function toDegrees(a){return (a/PI)*180}
function sin(a){return Math.sin(a)}
function cos(a){return Math.cos(a)}
function tg(a){return Math.tg(a)}
function atan(a){return Math.atan(a)}
function atan2(y,x){return Math.atan2(y,x)}
function abs(a){return Math.abs(a)}
function sqr(a){return Math.sqr(a)}
function sqrt(a){return Math.pow(a, 0.5)}
function pow(a, b){return Math.pow(a, b)}
function root(a, b){return Math.pow(a, 1.0/b)}
function rand(from, to){
    return Math.round(from+(Math.random()*(to-from)))
}
function round(a, p){
    var TMP=Math.round(a/p)
    return TMP*p
}
function log(a){return Math.log(a)}
function log10(a){return Math.log(a)/Math.log(10)}
function log2(a){return Math.log(a)/Math.log(2)}
function logx(a, x){return Math.log(a)/Math.log(x)}
function floor(a){return Math.floor(a)}
function ceil(a){return Math.ceil(a)}

function newArray(n){
    var T=new Array(n)
    return T
}

function toChar(a){
    String.fromCharCode(a)
}
function toInt(a){
    return parseInt(a)
}
function toFloat(a){
    return parseFloat(a)
}
function toLogic(a){
    if(a==true) return true
    if(a!=0) return true
    return false
}
function str(a){
    return ""+a
}

function getValueOf(n){
    if(typeof(n) == "undefined")
        return "undef"
    if(typeof(n) == "object")
        if(typeof(n.length) == "number"){
            var strasdfghjkl="["
            var l=n.length
            if(typeof(l)!="number")
                l=n.length()
            for(asdfghjkl=0; asdfghjkl<l; asdfghjkl++){
                strasdfghjkl+=getValueOf(n[asdfghjkl])
                if(asdfghjkl+1<n.length)
                    strasdfghjkl+=", "
            }
            strasdfghjkl+="]"
            return strasdfghjkl
        }
        else if(typeof(n.length()) == "number"){
            var strasdfghjkl="["
            for(asdfghjkl=0; asdfghjkl<n.length(); asdfghjkl++){
                strasdfghjkl+=getValueOf(n[asdfghjkl])
                if(asdfghjkl+1<n.length)
                    strasdfghjkl+=", "
            }
            strasdfghjkl+="]"
            return strasdfghjkl
        }
    return n
}
